Framework/Algorithms/src/ExtractSpectra.cpp

```diff
@@ -61,6 +60,22 @@ std::map<std::string, std::string> ExtractSpectra::validateInputs() {
       helpMessages["XMax"] = "XMax must be greater than XMin";
     }
   }
+
+  if (!isDefault("StartWorkspaceIndex")) {
+    const int minSpec_i = getProperty("StartWorkspaceIndex");
+    auto minSpec = static_cast<size_t>(minSpec_i);
+    API::MatrixWorkspace_sptr ws = getProperty("InputWorkspace");
+    const size_t numberOfSpectra = ws->indexInfo().globalSize();
+    int maxSpec_i = getProperty("EndWorkspaceIndex");
+    auto maxSpec = static_cast<size_t>(maxSpec_i);
+    if (isEmpty(maxSpec_i))
+      maxSpec = numberOfSpectra - 1;
+    if (maxSpec < minSpec) {
+      helpMessages["StartWorkspaceIndex"] = "StartWorkspaceIndex must be less than or equal to EndWorkspaceIndex";
```

Comment:
is it worth making this a warning and then swapping the indices?

Reply:
This is the behaviour that was present before, I've just moved it. I think raising this as an error is in line with what most Mantid algorithms do, so my opinion would be to keep it as is.